### PR TITLE
Issue:512 - Refresh the UI, since while initialising the controls it would have missed some debug events

### PR DESCRIPTION
**Pull request by vishnusarathashling**
Refresh the UI, since while initialising the controls it would have missed some debug events.
https://github.com/eclipse-embed-cdt/eclipse-plugins/issues/512.

### CODE REVIEW

File: plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java [hunk 1]
```diff
@@ -228,6 +228,8 @@ public void menuAboutToShow(IMenuManager manager) {
 		addDebugEventListener();
 		addToSyncService();
 		trackTreeSelectionChanges();
+		// Refresh the UI, since while initialising the controls it would have missed some debug events
```
Comment (ilg-ul) [inline plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java:231]: @jonahgraham, what is the current max line length for CDT? Does this comment line comply?

**Comment by ilg-ul** [comment]
@vishnusarathashling, the patch should be submitted agains the `develop` branch; it'll be merged into `master` a bit later, when the release will be prepared.

File: plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java (hunk 1 above)
Comment (vishnusarathashling) [reply plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java]: @ilg-ul I just splitted that into two lines.

**Comment by vishnusarathashling** [comment]
> @vishnusarathashling, the patch should be submitted agains the develop branch; it'll be merged into master a bit later, when the release will be prepared.

Done.

File: plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java (hunk 1 above)
Comment (ilg-ul) [reply plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java]: Ok, thank you.

Comment (jonahgraham) [reply plugins/org.eclipse.embedcdt.debug.gdbjtag.ui/src/org/eclipse/embedcdt/internal/debug/gdbjtag/ui/preferences/PeripheralRendering.java]: Thanks @vishnusarathashling for splitting it.

IIRC we try to keep < 120 chars for everything, including comments.

**Comment by jonahgraham** [comment]
Thank you @vishnusarathashling - well done on your first PR merged into the repo! I look forward to many more 😊

**Comment by vishnusarathashling** [comment]
> Thank you @vishnusarathashling - well done on your first PR merged into the repo! I look forward to many more 😊

Thanks Jonah!

**Comment by ilg-ul** [comment]
@vishnusarathashling, your patch should be available in v6.2.1, just released.

Please confirm that upgrading Eclipse fixes your problem.

**Comment by vishnusarathashling** [comment]
Sure @ilg-ul I will test and update.

**Comment by vishnusarathashling** [comment]
@ilg-ul Since I am little busy for 2 weeks, not sure I will be able to test this up quickly. We need to have some setups to be done to test this scenario. I will do it and update once I have it ready.

**Comment by ilg-ul** [comment]
Normally it takes a few minutes to install a fresh Eclipse and update it, using the default procedure; you don't have to do anything special, the new plug-ins release should be available for all existing Eclipses.

**Comment by vishnusarathashling** [comment]
Yeah got that, but to test the original issue we mentioned, it takes some setup to be done. Only in a particular board + configuration we were able to reproduce that. I will try to update this ASAP.

**Comment by vishnusarathashling** [comment]
I verified this and it's working with new embed-CDT.

**Comment by jonahgraham** [comment]
> I verified this and it's working with new embed-CDT.

Excellent - thanks @vishnusarathashling for confirming the change integrated properly and resolved the issue!

**Comment by vishnusarathashling** [comment]
Thanks @jonahgraham @ilg-ul .

**Comment by ilg-ul** [comment]
You're welcome! Thank you for your contribution.

**Pull request merged**